device/device.go

```diff
@@ -0,0 +1,99 @@
+package device
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/radio"
+)
+
+type SupportedDevice string
+
+const (
+	NRF51822 SupportedDevice = "NRF51822"
+	ESP8266  SupportedDevice = "ESP8266"
+	MICROBIT SupportedDevice = "MicroBit"
+)
+
+func Create(d SupportedDevice) DeviceInterface {
+	switch d {
```

Comment:
Surely we're defeating the purpose a bit of having everything abstracted out if we then go ahead and hardcode device information into the core package? I'm not sure all the type wrangling is worth it if we're just resorting to a switch after all?